### ISSUE TITLE
Idea for a new way to handle conversations

### ISSUE THREAD

**Issue by Kotov584** [vendelieu/telegram-bot#42]
**Is your feature request related to a problem? Please describe.**
I'm always frustrated when I have to define conversations using input handlers.
When you are using input handlers, it's pretty hard to write and manage complex conversations.

**Describe the solution you'd like**
Instead of input handlers, we could use classes for defining conversations. So we are able to write more well-architected code, even when it is complex.

Example: 

```kotlin
package com.example.blank.conversation

import eu.vendeli.tgbot.TelegramBot
import eu.vendeli.tgbot.api.message
import eu.vendeli.tgbot.types.User
import eu.vendeli.tgbot.types.internal.ProcessedUpdate

sealed class GreetingsConversation {
    object Start : GreetingsConversation() {
        fun handle(update: ProcessedUpdate, user: User, bot: TelegramBot) {
            message("Hello, let's move to second step!").send(user, bot)
            //set secondStep state
           //maybe something like 
           setData(user, "hello" to update.text)
           setState(user, GreetingsConversation::start)
        }
    }

    object SecondStep : GreetingsConversation() {
        fun handle(update: ProcessedUpdate, user: User, bot: TelegramBot) {
            // retrieve data from start step and proceed
            message("Nice job! You completed this.").send(user, bot)
            deleteState(user, GreetingsConversation)
        }
    }
}
```

**Additional context**
Not sure, if we can do this way.
We discussed about it in Telegram group.

**Comment by Kotov584** [comment]
or also this 
```kotlin
package com.example.blank.conversation

import eu.vendeli.tgbot.TelegramBot
import eu.vendeli.tgbot.api.message
import eu.vendeli.tgbot.types.User

class GreetingsConversation {
    fun start(user: User, bot: TelegramBot) {
        message("Hello, let's move to second step!").send(user, bot)
    }

    fun secondStep(user: User, bot: TelegramBot) {
        message("Nice job! You completed this.").send(user, bot)
    }
}
```

**Comment by vendelieu** [comment]
Experimental implementation is released in #61

**Comment by vendelieu** [comment]
implemented in #75
